FWCore/Framework/interface/es_Label.h

```diff
@@ -29,71 +30,75 @@
-         explicit L(std::unique_ptr<T> iP) : product_(std::move(iP)) {}
-         explicit L(T* iP) : product_(iP) {}
-         
-         T& operator*() { return *product_;}
-         T* operator->() { return product_.get(); }
-         mutable std::shared_ptr<T> product_;
-      };
-      template<int ILabel,typename T>
-         L<T,ILabel> l(std::shared_ptr<T>& iP) { 
-            L<T,ILabel> temp(iP);
-            return temp;
-         }
-      struct Label {
-         Label() : labels_(), default_() {}
-         Label(const char* iLabel) : labels_(), default_(iLabel) {}
-         Label(const std::string& iString) : labels_(), default_(iString) {}
-         Label(const std::string& iString, unsigned int iIndex) : 
-           labels_(iIndex+1,def()), default_() {labels_[iIndex] = iString;}
-         
-         Label& operator()(const std::string& iString, unsigned int iIndex) {
-            if(iIndex==labels_.size()){
-               labels_.push_back(iString);
-            } else if(iIndex > labels_.size()) {
-               std::vector<std::string> temp(iIndex+1,def());
-               copy_all(labels_, temp.begin());
-               labels_.swap(temp);
-            } else {
-               if( labels_[iIndex] != def() ) {
-                  Exception e(errors::Configuration,"Duplicate Label");
-                  e <<"The index "<<iIndex<<" was previously assigned the label \""
-                    <<labels_[iIndex]<<"\" and then was later assigned \""
-                    <<iString<<"\"";
-                  e.raise();
-               }
-               labels_[iIndex] = iString;
-            }
-            return *this;
-         }
-         Label& operator()(int iIndex, const std::string& iString) {
-            return (*this)(iString, iIndex);
-         }
-         
-         static const std::string& def() {
-            static const std::string s_def("\n\t");
-            return s_def;
-         }
-         std::vector<std::string> labels_;
-         std::string default_;
-      };
-      
-      inline Label label(const std::string& iString, int iIndex) {
-         return Label(iString, iIndex);
-      }
-      inline Label label(int iIndex, const std::string& iString) {
-         return Label(iString, iIndex);
+namespace edm::es {
+
+  template<typename T, int ILabel>
+  struct L {
+    using element_type = T;
+
+    L() = default;
+    explicit L(std::shared_ptr<T> iP) : product_{std::move(iP)} {}
+    explicit L(std::unique_ptr<T> iP) : product_{std::move(iP)} {}
+    explicit L(T* iP) : product_(iP) {}
+
+    T& operator*() { return *product_;}
+    T* operator->() { return product_.get(); }
+    mutable std::shared_ptr<T> product_{nullptr};
+  };
+
+  template<int ILabel, typename T>
+  L<T,ILabel> l(std::shared_ptr<T>& iP) {
+    return L<T,ILabel>{iP};
+  }
+
+  struct Label {
+    Label() = default;
+    Label(std::string_view const iString) : default_{iString} {}
```

Comment:
Was `string_view` the only change?

Reply:
That and the namespace collapsing.

Reply:
I think the switch to std::string_view is causing the compilation problem in HcalHardcodeGeometryEP.

Reply:
Yes, just confirmed that.  Will revert.